src/App.tsx

```diff
@@ -1,19 +1,96 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  FC,
+  useContext,
+  useState,
+  useEffect,
+  useReducer,
+  useMemo,
+  useRef,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { SortType } from './types/Filter';
+
+function filtTodos(
+  todos: Todo[],
+  sortType: SortType,
+) {
+  const visibleTodos = [...todos];
+
+  switch (sortType) {
+    case SortType.Active:
+      return visibleTodos.filter(todo => !todo.completed);
+
+    case SortType.Completed:
+      return visibleTodos.filter(todo => todo.completed);
+    default:
+      return visibleTodos;
+  }
+}
+
+const reducer = (count: number, action: string) => {
+  switch (action) {
+    case 'increase':
+      return count + 1;
+    case 'decrease':
+      return count - 1;
+    case 'clear':
+      return 0;
+    default:
+      return count;
+  }
+};
 
-export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const App: FC = () => {
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [errorMessage, setErrorMessage] = useState<string>('');
```

Comment:
No need to write styles for primitives.
```suggestion
  const [errorMessage, setErrorMessage] = useState('');
```